### PR TITLE
104 x dev nano diff

### PR DESCRIPTION
**Pull request by vischia**
Don't review yet, still working on it.

### CODE REVIEW

File: TTHAnalysis/macros/prepareEventVariablesFriendTree.py [hunk 1]
```diff
@@ -478,11 +483,11 @@ def analyze(self,event):
                 cmd = "echo \"{base} -d {data} -c {chunk} {post}\" | {super} {writelog}".format(super=super, writelog=writelog, base=basecmd, data=name, chunk=chunk, post=friendPost)
             elif options.env == "oviedo":
                 cmd = "{super} {writelog} {base} -d {data} -c {chunk} {post} ".format(super=super, writelog=writelog, base=basecmd, data=name, chunk=chunk, post=friendPost)
-            if options.queue == "cp3" and options.env == "uclouvain":
+            if options.queue == "cp3" and ("uclouvain" in options.env):
                 full_subfile = "{subfile}{data}_{chunk}.sh".format(subfile=options.subfile, data=name, chunk=chunk)
-                subfile = open(full_subfile, "w")
+                subfile = open(full_subfile, "w") 
                 subfile.write("""#! /bin/bash
-#SBATCH --ntasks=8
```
Comment (vischia) [inline TTHAnalysis/macros/prepareEventVariablesFriendTree.py:485]: This breaks our batch, maybe we can put a conditional

File: TTHAnalysis/macros/prepareEventVariablesFriendTree.py [hunk 2]
```diff
@@ -504,11 +509,11 @@ def analyze(self,event):
...
             elif options.env == "oviedo":
                 cmd = "{super} {base} -d {data} {post} {writelog}".format(super=super, writelog=writelog, base=basecmd, data=name, chunk=chunk, post=friendPost)
-            if options.queue == "cp3" and options.env == "uclouvain":
+            if options.queue == "cp3" and ("uclouvain" in options.env):
                 full_subfile = "{subfile}{data}_{chunk}.sh".format(subfile=options.subfile, data=name, chunk=chunk)
-                subfile = open(full_subfile, "w")
+                subfile = open(full_subfile, "w")#--ntasks=2
                 subfile.write("""#! /bin/bash
-#SBATCH --ntasks=8
+#SBATCH
```
Comment (vischia) [inline TTHAnalysis/macros/prepareEventVariablesFriendTree.py:516]: This breaks our batch, maybe we can put a conditional

**Comment by vischia** [comment]
OK I added a couple comments and it should be more or less cleaned up

File: TTHAnalysis/macros/prepareEventVariablesFriendTree.py (hunk 2 above)
Comment (vischia) [reply TTHAnalysis/macros/prepareEventVariablesFriendTree.py:516]: @sscruz

File: TTHAnalysis/macros/prepareEventVariablesFriendTree.py (hunk 1 above)
Comment (vischia) [reply TTHAnalysis/macros/prepareEventVariablesFriendTree.py:485]: @sscruz

**Comment by vischia** [comment]
Dear Sergio, 

I hope this message finds you well.

Plis riviú

File: TTHAnalysis/macros/prepareEventVariablesFriendTree.py (hunk 1 above)
Comment (sscruz) [reply TTHAnalysis/macros/prepareEventVariablesFriendTree.py:485]: do you need to hardcode the names of the partitions? there's a -q option for that :)

File: TTHAnalysis/macros/prepareEventVariablesFriendTree.py (hunk 2 above)
Comment (sscruz) [reply TTHAnalysis/macros/prepareEventVariablesFriendTree.py:516]: do you need to write a script for each job? you can use lxbatch_runner.sh, see how it's done for --env psi

File: TTHAnalysis/python/plotter/ttH-multilepton/mca-includes/mca-2lss-sigprompt.txt [hunk 3]
```diff
@@ -1,4 +1,5 @@
 ttH+    : TTHnobb_fxfx : xsec : LepGood1_isMatchRightCharge && LepGood2_isMatchRightCharge; FillColor=ROOT.kOrange+10
+#ttH+    : TTHnobb_pow : 0.5071*0.418 : LepGood1_isMatchRightCharge && LepGood2_isMatchRightCharge; FillColor=ROOT.kOrange+10 #added by H to run on v5
```
Comment (sscruz) [inline TTHAnalysis/python/plotter/ttH-multilepton/mca-includes/mca-2lss-sigprompt.txt]: Please remove the commented line

File: TTHAnalysis/python/plotter/ttH-multilepton/ttH_plots.py [hunk 4]
```diff
@@ -125,6 +120,8 @@ def doprescale3l(x,torun):
 
     if '2lss_' in torun:
         x = base('2lss')
+        if 'diff' in torun:
+            x = x.replace('2lss_3l_plots.txt', '2lss_3l_plots_diff.txt').replace('--showMCError', '-X --showMCError')
```
Comment (sscruz) [inline TTHAnalysis/python/plotter/ttH-multilepton/ttH_plots.py:122]: can't you do this with --xP
 and --sP?

File: TTHAnalysis/python/plotter/ttH-multilepton/ttH_plots.py [hunk 5]
```diff
@@ -30,7 +30,9 @@
     P0     = "/pool/ciencias/HeppyTrees/EdgeZ/TTH/"
 
 if 'cism.ucl.ac.be' in os.environ['HOSTNAME']:
-    P0 = "/nfs/user/pvischia/tth/v5pre/"
+    Tag =sys.argv[3] #2lss_diff_NoTop-tagged or 2lss_diff_Top-tagged
```
Comment (sscruz) [inline TTHAnalysis/python/plotter/ttH-multilepton/ttH_plots.py]: is Tag used anywhere else in the code?

File: TTHAnalysis/python/tools/higgsDiffRegressionTTH_jagged.py [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+            self.out.fillBranch('%sHTXS_Higgs_pt%s'%(self.label,jesLabel), getattr(event,"HTXS_Higgs_pt"))
+            self.out.fillBranch('%sHTXS_Higgs_y%s' %(self.label,jesLabel), getattr(event,"HTXS_Higgs_y") )
+            # I must patch these two to fill only for TTH, otherwise the friend does not exist etc. Maybe produce friend also for background
+            self.out.fillBranch('%sHgen_vis_pt%s'  %(self.label,jesLabel), getattr(event,'Hreco_pTTrueGen'))
+            self.out.fillBranch('%sHgen_tru_pt%s'  %(self.label,jesLabel), getattr(event,'Hreco_pTTrueGenPlusNu')) # the same as HTXS_Higgs_pt
+
+        return True
+
+higgsDiffRegressionTTH = lambda : HiggsDiffRegressionTTH(label='Hreco_',
+                                                         btagDeepCSVveto = 'M')
```
Comment (sscruz) [inline TTHAnalysis/python/tools/higgsDiffRegressionTTH_jagged.py]: You have 3 or 4 modules that more or less do the same. Please consider making one clean module

File: TTHAnalysis/python/tools/nanoAOD/ttH_modules.py [hunk 7]
```diff
@@ -324,6 +324,7 @@ def _fires(ev, path):
 
 BDThttTT_Hj = lambda : BDT_eventReco(os.environ["CMSSW_BASE"]+'/src/CMGTools/TTHAnalysis/data/kinMVA/tth/TMVAClassification_bloose_BDTG.weights.xml',
                                      os.environ["CMSSW_BASE"]+'/src/CMGTools/TTHAnalysis/data/kinMVA/tth/TMVAClassification_btight_BDTG.weights.xml',
+#                                    os.environ["CMSSW_BASE"]+'/src/CMGTools/TTHAnalysis/data/kinMVA/tth/Hj_2017_configA_dcsv_BDTG.weights.xml',
```
Comment (sscruz) [inline TTHAnalysis/python/tools/nanoAOD/ttH_modules.py]: Please remove commented line

File: TTHAnalysis/macros/prepareEventVariablesFriendTree.py (hunk 1 above)
Comment (vischia) [reply TTHAnalysis/macros/prepareEventVariablesFriendTree.py:485]: Unfortunately for UCLouvain yes :(

File: TTHAnalysis/macros/prepareEventVariablesFriendTree.py (hunk 2 above)
Comment (vischia) [reply TTHAnalysis/macros/prepareEventVariablesFriendTree.py:516]: Unfortunately in uclouvain yes, the way it's done in lxbatch_runner doesn't work :(

File: TTHAnalysis/python/plotter/ttH-multilepton/mca-includes/mca-2lss-sigprompt.txt (hunk 3 above)
Comment (vischia) [reply TTHAnalysis/python/plotter/ttH-multilepton/mca-includes/mca-2lss-sigprompt.txt]: Whops, I missed that. Removed.

File: TTHAnalysis/python/plotter/ttH-multilepton/mca-includes/mca-2lss-sigprompt.txt [hunk 8]
```diff
@@ -23,7 +24,7 @@ Rares: VHToNonbb : xsec : LepGood1_isMatchRightCharge && LepGood2_isMatchRightCh
 Rares: qqHZZ4L : xsec : (LepGood1_isMatchRightCharge && LepGood2_isMatchRightCharge) ; FillColor=ROOT.kAzure-9
 Rares: TTTT : xsec : LepGood1_isMatchRightCharge && LepGood2_isMatchRightCharge ; FillColor=ROOT.kAzure-9, years="2016\,2018"
 Rares: TTTT+TTTT_PS : xsec : LepGood1_isMatchRightCharge && LepGood2_isMatchRightCharge ; FillColor=ROOT.kAzure-9, years="2017"
-Rares: TTTW : xsec : LepGood1_isMatchRightCharge && LepGood2_isMatchRightCharge ; FillColor=ROOT.kAzure-9
+#Rares: TTTW : xsec : LepGood1_isMatchRightCharge && LepGood2_isMatchRightCharge ; FillColor=ROOT.kAzure-9
```
Comment (vischia) [inline TTHAnalysis/python/plotter/ttH-multilepton/mca-includes/mca-2lss-sigprompt.txt]: Not sure of what to do here, I can't remember why is it commented. Maybe we had not copied the sample at the time?

File: TTHAnalysis/python/tools/nanoAOD/ttH_modules.py (hunk 7 above)
Comment (vischia) [reply TTHAnalysis/python/tools/nanoAOD/ttH_modules.py]: Done

File: TTHAnalysis/python/tools/higgsDiffRegressionTTH_jagged.py (hunk 6 above)
Comment (vischia) [reply TTHAnalysis/python/tools/higgsDiffRegressionTTH_jagged.py]: This is a design choice (one will be superstable, the others will keep being changed. Also, speed.

I now removed higgsDiffRegressionTTH_jagged.py, that's not useful anymore.

File: TTHAnalysis/python/plotter/ttH-multilepton/ttH_plots.py (hunk 5 above)
Comment (vischia) [reply TTHAnalysis/python/plotter/ttH-multilepton/ttH_plots.py]: That's a remnant of Hesham's code I think. I removed it together with the commented line.

File: TTHAnalysis/python/plotter/ttH-multilepton/ttH_plots.py (hunk 4 above)
Comment (vischia) [reply TTHAnalysis/python/plotter/ttH-multilepton/ttH_plots.py:122]: Theoretically yes, but in the spirit of ttH_plots.py it's handier to not have to add too many command line options (that's the whole point of the ttH_plots.py. Also, probably recommended by the PubComm . Left it as it is.

File: TTHAnalysis/python/plotter/ttH-multilepton/mca-includes/mca-2lss-sigprompt.txt (hunk 8 above)
Comment (vischia) [reply TTHAnalysis/python/plotter/ttH-multilepton/mca-includes/mca-2lss-sigprompt.txt]: I removed it, it's clearly wrong (samples exist)

**Comment by vischia** [comment]
Resolved all review comments, plis merge :heart:

**Comment by vischia** [comment]
Hi @sscruz , can you please merge?

File: TTHAnalysis/macros/prepareEventVariablesFriendTree.py (hunk 2 above)
Comment (sscruz) [reply TTHAnalysis/macros/prepareEventVariablesFriendTree.py:516]: then i would rather change lxbatch_runner instead and avoid producing n files, but ok, i'm not gonna use it myself :)

**Review by sscruz — APPROVED** [review]

**Pull request merged**